@font-face src must serialize specified URLs, not URLs resolved against the base

darinadler opened the pull request:
#### 443de8e8e9716b164f0a25da35e9235efd72d4dd
<pre>
@font-face src must serialize specified URLs, not URLs resolved against the base
<a href="https://bugs.webkit.org/show_bug.cgi?id=247547">https://bugs.webkit.org/show_bug.cgi?id=247547</a>
rdar://problem/102012585

Reviewed by Sam Weinig.

* LayoutTests/fast/css/font-face-src-parsing-expected.txt: Updated to expect successful
parsing of all the values for src. The old code expected some to be skipped, and
they are not skipped any more.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/font-family-src-quoted-expected.txt:
Expect PASS.

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources): Updated since CSSFontFaceSrcLocalValue and
CSSFontFaceSrcResourceValue are now two separate classes and CSSFontFaceSource gets the URL from
the FontLoadRequest rather than taking a separate URL argument.

* Source/WebCore/css/CSSFontFaceSet.cpp:
(WebCore::CSSFontFaceSet::ensureLocalFontFacesForFamilyRegistered): Take AtomString.
(WebCore::CSSFontFaceSet::addToFacesLookupTable): Use AtomString.
* Source/WebCore/css/CSSFontFaceSet.h: Updated for above.

* Source/WebCore/css/CSSFontFaceSource.cpp:
(WebCore::CSSFontFaceSource::CSSFontFaceSource): Updated the argument name since this now takes
only a font face name, not a URL. Also, a font face name is *not* the same thing as a family name,
so don&apos;t call it a family name. Change argument types for things we take ownership of to the
appropriate owning type: AtomString and UniqueRef.
(WebCore::CSSFontFaceSource::opportunisticallyStartFontDataURLLoading): Check the length of URL
from the font request itself to decide if it&apos;s short enough, rather than using m_familyNameOrURI.
(WebCore::CSSFontFaceSource::fontLoaded): Removed the argument from ensureCustomFontData. We don&apos;t
need to pass the URL back to the object which supplied the URL in the first place.
(WebCore::CSSFontFaceSource::load): Updated for the name change of m_fontFaceName.
(WebCore::CSSFontFaceSource::font): Ditto. Also removed the argument from ensureCustomFontData
and createFont.

* Source/WebCore/css/CSSFontFaceSource.h: Changed the constructors: We don’t need to pass in a
URL along with a FontLoadRequest, which contains the URL. The font face name is not the
same thing as a family name, so let&apos;s not call it that.  Change argument types for things we take
ownership of to the appropriate owning type: AtomString and UniqueRef. There is not font face name
or URL when creating from an ArrayBufferView, so don&apos;t require the caller to pass a null one in.
Removed the familyNameOrURI function, which was unneeded. Replaced m_familyNameOrURI with
m_fontFaceName. Made m_fontRequest const to emphasize it&apos;s never changed after construction.

* Source/WebCore/css/CSSFontFaceSrcValue.cpp: Broke the CSSFontFaceSrcValue into two classes,
CSSFontFaceSrcLocalValue and CSSFontFaceSrcResourceValue. Otherwise this class would have mostly
been variant processing, since there are basically no data members in common.
(WebCore::CSSFontFaceSrcLocalValue::CSSFontFaceSrcLocalValue): Added.
(WebCore::CSSFontFaceSrcLocalValue::create): Added.
(WebCore::CSSFontFaceSrcLocalValue::svgFontFaceElement const): Added.
(WebCore::CSSFontFaceSrcLocalValue::setSVGFontFaceElement): Added.
(WebCore::CSSFontFaceSrcLocalValue::customCSSText const): Added.
(WebCore::CSSFontFaceSrcLocalValue::equals const): Added.
(WebCore::CSSFontFaceSrcResourceValue::CSSFontFaceSrcResourceValue): Added. Stores the location
as a ResolvedURL, the same way CSSImageValue does.
(WebCore::CSSFontFaceSrcResourceValue::create): Added.
(WebCore::CSSFontFaceSrcResourceValue::fontLoadRequest): Added. Besides what the old version
of this function did, this now checks if the format is one we can load and returns nullptr if not.
(WebCore::CSSFontFaceSrcResourceValue::customTraverseSubresources const): Added.
(WebCore::CSSFontFaceSrcResourceValue::customCSSText const): Added. Here is where the core of the
bug fix is: we serialize the specified URL string, not the resolved URL.
(WebCore::CSSFontFaceSrcResourceValue::equals const): Added.

* Source/WebCore/css/CSSFontFaceSrcValue.h: Broke the CSSFontFaceSrcValue class into two classes,
CSSFontFaceSrcLocalValue and CSSFontFaceSrcResourceValue. Removed unneeded functions resource,
format, isLocal, setFormat, isSupportedFormat, isSVGFontFaceSrc, abd isSVGFontTarget. Added
fontFaceName and isEmpty functions to CSSFontFaceSrcLocalValue. Changed CSSFontFaceSrcResourceValue
to take and store a ResolvedURL. Added isEmpty and isFormatSVG functions to
CSSFontFaceSrcResourceValue. Changed fontLoadRequest to take a ScriptExecutionContext&amp; and also
removed the isSVG argument since CSSFontFaceSrcResourceValue itself is the where the answer to that
question comes from.

* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived): Updated for the two classes that replace CSSFontFaceSrcValue.
* Source/WebCore/css/CSSValue.h: Ditto.

* Source/WebCore/css/FontFace.cpp:
(WebCore::populateFontFaceWithArrayBuffer): Remove unneeded null URL string argument.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::consumePaintStroke): Updated to call consumeURL.
(WebCore::consumeNoneOrURI): Ditto.
(WebCore::consumePathOperation): Ditto.
(WebCore::consumeGridTrackRepeatFunction): Updated to use a rangee-based for loop.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeURLRaw): Renamed from consumeUrlAsStringView.
(WebCore::CSSPropertyParserHelpers::consumeURL): Renamed from consumeUrl.
(WebCore::CSSPropertyParserHelpers::consumeFilter): Updated to call consumeURL.
(WebCore::CSSPropertyParserHelpers::consumeImage): Ditto.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h: Updated for the name changes.

* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrcURI): Use the ResolvedURL
instead of immediately turning it into a string. Call the new
CSSFontFaceSrcResourceValue::create.
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrcLocal): Call the new
CSSFontFaceSrcLocalValue::create.

* Source/WebCore/css/parser/CSSTokenizer.cpp:
(WebCore::CSSTokenizer::consumeIdentLikeToken): Use consumeURLToken.
(WebCore::CSSTokenizer::consumeURLToken): Renamed from consumeUrlToken.
* Source/WebCore/css/parser/CSSTokenizer.h: Updated for the name change.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::fontLoadRequest): Changed the URL argument to a const String&amp;.
It was String&amp; before by accident.
* Source/WebCore/dom/Document.h: Ditto.
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::fontLoadRequest): Ditto.
* Source/WebCore/dom/ScriptExecutionContext.h: Ditto.

* Source/WebCore/loader/FontLoadRequest.h: Took out the unnneded remoteURI arguments
from the ensureCustomFontData and createFont functions. These involve the font load
request passing the URI back to the object it came from, so they aren&apos;t needed.

* Source/WebCore/loader/cache/CachedFont.cpp:
(WebCore::CachedFont::ensureCustomFontData): Removed remoteURI argument.
(WebCore::CachedFont::createFont): Ditto.
* Source/WebCore/loader/cache/CachedFont.h: Ditto.
* Source/WebCore/loader/cache/CachedFontLoadRequest.h: Ditto.
* Source/WebCore/loader/cache/CachedSVGFont.cpp:
(WebCore::CachedSVGFont::createFont): Ditto.
(WebCore::CachedSVGFont::ensureCustomFontData): Ditto.
(WebCore::CachedSVGFont::maybeInitializeExternalSVGFontElement): Ditto.
The logic here got simpler becuase when we have an already-parsed URL instead of
just an atom string we can take advantage of member functions.
(WebCore::CachedSVGFont::firstFontFace): Ditto.
* Source/WebCore/loader/cache/CachedSVGFont.h: Ditto.

* Source/WebCore/svg/SVGFontFaceElement.cpp:
(WebCore::SVGFontFaceElement::rebuildFontFace): Use CSSFontFaceSrcLocalValue::create,
a range-based for loop, and pass a reference instead of a pointer. Call createSrcValue.

* Source/WebCore/svg/SVGFontFaceNameElement.cpp:
(WebCore::SVGFontFaceNameElement::createSrcValue const): Renamed from srcValue, and return a
CSSFontFaceSrcLocalValue.
* Source/WebCore/svg/SVGFontFaceNameElement.h: Ditto.

* Source/WebCore/svg/SVGFontFaceSrcElement.cpp:
(WebCore::SVGFontFaceSrcElement::createSrcValue const): Renamed from srcValue, and updated to work
with the two different createSrcValue functions that now have different return types.

* Source/WebCore/svg/SVGFontFaceUriElement.cpp:
(WebCore::SVGFontFaceUriElement::createSrcValue const): Renamed from srcValue, and return a
CSSFontFaceSrcResourceValue.
(WebCore::isSVGFontTarget): Check if the font is an SVG font without creating and destroying a
CSSFontFaceSrcValue, which is overkill since we can check the format attribute directly.
* Source/WebCore/svg/SVGFontFaceUriElement.h: Return a CSSFontFaceSrcResourceValue.

* Source/WebCore/workers/WorkerFontLoadRequest.cpp:
(WebCore::WorkerFontLoadRequest::ensureCustomFontData): Removed remoteURI argument.
(WebCore::WorkerFontLoadRequest::createFont): Ditto.
* Source/WebCore/workers/WorkerFontLoadRequest.h: Ditto.

* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::fontLoadRequest): Changed the URL argument to a const String&amp;.
It was String&amp; before by accident.
* Source/WebCore/workers/WorkerGlobalScope.h: Ditto.

Canonical link: <a href="https://commits.webkit.org/256387@main">https://commits.webkit.org/256387@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3719250ddaa4aee8455f17f1941765cb450455c6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/95636 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/4897 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/28684 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/105214 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/165511 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/99621 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/4962 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/33659 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/88008 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/101063 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/101297 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/3640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/82253 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/30693 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/85509 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/87412 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/73529 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/39386 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/18960 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/37085 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/20268 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4414 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/41081 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/42920 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/43069 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/39517 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

weinig commented on Source/WebCore/css/CSSFontFaceSource.cpp:
```diff
@@ -94,17 +93,16 @@ CSSFontFaceSource::CSSFontFaceSource(CSSFontFace& owner, const String& familyNam
 
-CSSFontFaceSource::CSSFontFaceSource(CSSFontFace& owner, const String& familyNameOrURI, SVGFontFaceElement& fontFace)
-    : m_familyNameOrURI(familyNameOrURI)
+CSSFontFaceSource::CSSFontFaceSource(CSSFontFace& owner, const String& fontFaceName, SVGFontFaceElement& fontFace)
```
Given this assigns to an AtomString m_fontFaceName, I would consider making this parameter an AtomString as well (in the off chance the caller has one). I would also make it take the parameter by value and WTFMove into the member.

darinadler replied:
Did that

weinig commented on Source/WebCore/svg/SVGFontFaceNameElement.h:
```diff
@@ -23,14 +23,14 @@
     static Ref<SVGFontFaceNameElement> create(const QualifiedName&, Document&);
     
-    Ref<CSSFontFaceSrcValue> srcValue() const;
+    Ref<CSSFontFaceSrcLocalValue> srcValue() const;
```
I think this could be more clearly something that creates a new allocation each time with a better name. Maybe throw in a "create"?

weinig reviewed (approved):
This change brings up two topics that have nothing to do with the actual meat of this change! 

1. I have been thinking a bit about pass-by-value and pass-by-reference as used in WebKit for ref-counted objects, and I think we overusing pass-by-reference quite a bit. As a rule, if function holds onto a parameter after it returns, the paramter should be pass-by-value. The logic being that the function is going to need to ref the object, so it might as well be done by the caller, allowing for them to optionally WTFMove() it in if it's their last reference. Likewise, as a rule if a function doesn't hold onto the parameter after return, it should be pass-by-reference, as that avoids refs. 

The tricky one is if a function can conditionally hold onto the object or not. In an ideal world, the compiler would do whole-program-analysis and figure out how to ellide the references for us, or we could be living the module future of tomorrow and feel comfortable just taking things by universal reference because there are no cpp files anymore, but alas. For these conditional ones, we can either one or implement two variants of the function (one r-value reference, one l-value reference), but I am not sure how often this conditional sitution actually happens.

As a somewhat interesting note/asside, with ARC in Swift (I am not sure if the Objc ARC has the same rules), the rule the Swift team adopted is that for init() functions, arguments are moved (that is, if the caller needs to hold onto a reference, the caller must ref, but if it is the caller's last use, they can avoid that ref, [of course, caller here is the compiler, not an actual person], and for all other functions, arguments are borrowed (that is, it is the callee that must ref the object if it wants to hold onto it). The theory behind this, as I understand it, is that it is more common for the initializer of a class to take arguments that it then holds onto, and it is more common for non-initializers to just use the parameter in passing (kind of a "generational hypothesis" style thought excersise). That intuition matches my experience, so it seems to be a good enough rule. Now, this kind of hard rule is needed in Swift because a) they don't have a way to annotate move or borrow yet, though that is coming, and b) the ABI needs to encode this information for libraries to work. We don't happen to have either of those constraints.


2. I really want to make the CSSValue subclasses into immutable value types. The biggest thing holding that back (other than time and the fact that I have a full time job doing something else) is that CSSFontFaceSrcValue (now CSSFontFaceSrcResourceValue) and CSSImageValue hold onto CachedResource handles. I posit that for CSSImageValue, the StyleCachedImage is probably a better home for the handle, and any sharing that is happening in via the CSSImageValue will just easily happen via the resource cache (that said, there are few other constraints that block that change like PageSerializer grabing data from CSSImageValue directly, but I figure I can figure out how to resolve those).

For CSSFontFaceSrcValue (now CSSFontFaceSrcResourceValue), is there a comparable spot outside of the CSSValue subclass itself where the CachedResourceHandle could live? Maybe in the CSSFontFace or CSSFontFaceSource?

darinadler committed:
@font-face src must serialize specified URLs, not URLs resolved against the base
https://bugs.webkit.org/show_bug.cgi?id=247547
rdar://problem/102012585

Reviewed by Sam Weinig.

* LayoutTests/fast/css/font-face-src-parsing-expected.txt: Updated to expect successful
parsing of all the values for src. The old code expected some to be skipped, and
they are not skipped any more.

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/font-family-src-quoted-expected.txt:
Expect PASS.

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources): Updated since CSSFontFaceSrcLocalValue and
CSSFontFaceSrcResourceValue are now two separate classes and CSSFontFaceSource gets the URL from
the FontLoadRequest rather than taking a separate URL argument.

* Source/WebCore/css/CSSFontFaceSet.cpp:
(WebCore::CSSFontFaceSet::ensureLocalFontFacesForFamilyRegistered): Take AtomString.
(WebCore::CSSFontFaceSet::addToFacesLookupTable): Use AtomString.
* Source/WebCore/css/CSSFontFaceSet.h: Updated for above.

* Source/WebCore/css/CSSFontFaceSource.cpp:
(WebCore::CSSFontFaceSource::CSSFontFaceSource): Updated the argument name since this now takes
only a font face name, not a URL. Also, a font face name is *not* the same thing as a family name,
so don't call it a family name. Change argument types for things we take ownership of to the
appropriate owning type: AtomString and UniqueRef.
(WebCore::CSSFontFaceSource::opportunisticallyStartFontDataURLLoading): Check the length of URL
from the font request itself to decide if it's short enough, rather than using m_familyNameOrURI.
(WebCore::CSSFontFaceSource::fontLoaded): Removed the argument from ensureCustomFontData. We don't
need to pass the URL back to the object which supplied the URL in the first place.
(WebCore::CSSFontFaceSource::load): Updated for the name change of m_fontFaceName.
(WebCore::CSSFontFaceSource::font): Ditto. Also removed the argument from ensureCustomFontData
and createFont.

* Source/WebCore/css/CSSFontFaceSource.h: Changed the constructors: We don’t need to pass in a
URL along with a FontLoadRequest, which contains the URL. The font face name is not the
same thing as a family name, so let's not call it that.  Change argument types for things we take
ownership of to the appropriate owning type: AtomString and UniqueRef. There is not font face name
or URL when creating from an ArrayBufferView, so don't require the caller to pass a null one in.
Removed the familyNameOrURI function, which was unneeded. Replaced m_familyNameOrURI with
m_fontFaceName. Made m_fontRequest const to emphasize it's never changed after construction.

* Source/WebCore/css/CSSFontFaceSrcValue.cpp: Broke the CSSFontFaceSrcValue into two classes,
CSSFontFaceSrcLocalValue and CSSFontFaceSrcResourceValue. Otherwise this class would have mostly
been variant processing, since there are basically no data members in common.
(WebCore::CSSFontFaceSrcLocalValue::CSSFontFaceSrcLocalValue): Added.
(WebCore::CSSFontFaceSrcLocalValue::create): Added.
(WebCore::CSSFontFaceSrcLocalValue::svgFontFaceElement const): Added.
(WebCore::CSSFontFaceSrcLocalValue::setSVGFontFaceElement): Added.
(WebCore::CSSFontFaceSrcLocalValue::customCSSText const): Added.
(WebCore::CSSFontFaceSrcLocalValue::equals const): Added.
(WebCore::CSSFontFaceSrcResourceValue::CSSFontFaceSrcResourceValue): Added. Stores the location
as a ResolvedURL, the same way CSSImageValue does.
(WebCore::CSSFontFaceSrcResourceValue::create): Added.
(WebCore::CSSFontFaceSrcResourceValue::fontLoadRequest): Added. Besides what the old version
of this function did, this now checks if the format is one we can load and returns nullptr if not.
(WebCore::CSSFontFaceSrcResourceValue::customTraverseSubresources const): Added.
(WebCore::CSSFontFaceSrcResourceValue::customCSSText const): Added. Here is where the core of the
bug fix is: we serialize the specified URL string, not the resolved URL.
(WebCore::CSSFontFaceSrcResourceValue::equals const): Added.

* Source/WebCore/css/CSSFontFaceSrcValue.h: Broke the CSSFontFaceSrcValue class into two classes,
CSSFontFaceSrcLocalValue and CSSFontFaceSrcResourceValue. Removed unneeded functions resource,
format, isLocal, setFormat, isSupportedFormat, isSVGFontFaceSrc, abd isSVGFontTarget. Added
fontFaceName and isEmpty functions to CSSFontFaceSrcLocalValue. Changed CSSFontFaceSrcResourceValue
to take and store a ResolvedURL. Added isEmpty and isFormatSVG functions to
CSSFontFaceSrcResourceValue. Changed fontLoadRequest to take a ScriptExecutionContext& and also
removed the isSVG argument since CSSFontFaceSrcResourceValue itself is the where the answer to that
question comes from.

* Source/WebCore/css/CSSValue.cpp:
(WebCore::CSSValue::visitDerived): Updated for the two classes that replace CSSFontFaceSrcValue.
* Source/WebCore/css/CSSValue.h: Ditto.

* Source/WebCore/css/FontFace.cpp:
(WebCore::populateFontFaceWithArrayBuffer): Remove unneeded null URL string argument.

* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::consumePaintStroke): Updated to call consumeURL.
(WebCore::consumeNoneOrURI): Ditto.
(WebCore::consumePathOperation): Ditto.
(WebCore::consumeGridTrackRepeatFunction): Updated to use a rangee-based for loop.

* Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp:
(WebCore::CSSPropertyParserHelpers::consumeURLRaw): Renamed from consumeUrlAsStringView.
(WebCore::CSSPropertyParserHelpers::consumeURL): Renamed from consumeUrl.
(WebCore::CSSPropertyParserHelpers::consumeFilter): Updated to call consumeURL.
(WebCore::CSSPropertyParserHelpers::consumeImage): Ditto.
* Source/WebCore/css/parser/CSSPropertyParserHelpers.h: Updated for the name changes.

* Source/WebCore/css/parser/CSSPropertyParserWorkerSafe.cpp:
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrcURI): Use the ResolvedURL
instead of immediately turning it into a string. Call the new
CSSFontFaceSrcResourceValue::create.
(WebCore::CSSPropertyParserHelpersWorkerSafe::consumeFontFaceSrcLocal): Call the new
CSSFontFaceSrcLocalValue::create.

* Source/WebCore/css/parser/CSSTokenizer.cpp:
(WebCore::CSSTokenizer::consumeIdentLikeToken): Use consumeURLToken.
(WebCore::CSSTokenizer::consumeURLToken): Renamed from consumeUrlToken.
* Source/WebCore/css/parser/CSSTokenizer.h: Updated for the name change.

* Source/WebCore/dom/Document.cpp:
(WebCore::Document::fontLoadRequest): Changed the URL argument to a const String&.
It was String& before by accident.
* Source/WebCore/dom/Document.h: Ditto.
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::ScriptExecutionContext::fontLoadRequest): Ditto.
* Source/WebCore/dom/ScriptExecutionContext.h: Ditto.

* Source/WebCore/loader/FontLoadRequest.h: Took out the unnneded remoteURI arguments
from the ensureCustomFontData and createFont functions. These involve the font load
request passing the URI back to the object it came from, so they aren't needed.

* Source/WebCore/loader/cache/CachedFont.cpp:
(WebCore::CachedFont::ensureCustomFontData): Removed remoteURI argument.
(WebCore::CachedFont::createFont): Ditto.
* Source/WebCore/loader/cache/CachedFont.h: Ditto.
* Source/WebCore/loader/cache/CachedFontLoadRequest.h: Ditto.
* Source/WebCore/loader/cache/CachedSVGFont.cpp:
(WebCore::CachedSVGFont::createFont): Ditto.
(WebCore::CachedSVGFont::ensureCustomFontData): Ditto.
(WebCore::CachedSVGFont::maybeInitializeExternalSVGFontElement): Ditto.
The logic here got simpler becuase when we have an already-parsed URL instead of
just an atom string we can take advantage of member functions.
(WebCore::CachedSVGFont::firstFontFace): Ditto.
* Source/WebCore/loader/cache/CachedSVGFont.h: Ditto.

* Source/WebCore/svg/SVGFontFaceElement.cpp:
(WebCore::SVGFontFaceElement::rebuildFontFace): Use CSSFontFaceSrcLocalValue::create,
a range-based for loop, and pass a reference instead of a pointer. Call createSrcValue.

* Source/WebCore/svg/SVGFontFaceNameElement.cpp:
(WebCore::SVGFontFaceNameElement::createSrcValue const): Renamed from srcValue, and return a
CSSFontFaceSrcLocalValue.
* Source/WebCore/svg/SVGFontFaceNameElement.h: Ditto.

* Source/WebCore/svg/SVGFontFaceSrcElement.cpp:
(WebCore::SVGFontFaceSrcElement::createSrcValue const): Renamed from srcValue, and updated to work
with the two different createSrcValue functions that now have different return types.

* Source/WebCore/svg/SVGFontFaceUriElement.cpp:
(WebCore::SVGFontFaceUriElement::createSrcValue const): Renamed from srcValue, and return a
CSSFontFaceSrcResourceValue.
(WebCore::isSVGFontTarget): Check if the font is an SVG font without creating and destroying a
CSSFontFaceSrcValue, which is overkill since we can check the format attribute directly.
* Source/WebCore/svg/SVGFontFaceUriElement.h: Return a CSSFontFaceSrcResourceValue.

* Source/WebCore/workers/WorkerFontLoadRequest.cpp:
(WebCore::WorkerFontLoadRequest::ensureCustomFontData): Removed remoteURI argument.
(WebCore::WorkerFontLoadRequest::createFont): Ditto.
* Source/WebCore/workers/WorkerFontLoadRequest.h: Ditto.

* Source/WebCore/workers/WorkerGlobalScope.cpp:
(WebCore::WorkerGlobalScope::fontLoadRequest): Changed the URL argument to a const String&.
It was String& before by accident.
* Source/WebCore/workers/WorkerGlobalScope.h: Ditto.

Canonical link: https://commits.webkit.org/256387@main
